### ISSUE TITLE
Package Manager needs updated cache of repos before installation

### ISSUE THREAD

**Issue by isc-kiyer** [intersystems/ipm#536]
The package manager keeps a cache of available versions of modules locally for each repo. This is useful for more performant dependency resolution. However, at time of installing a module, this cache should be updated so that it is working against latest info. Currently the cache is only updated when the repo is created or if manually forced to update using the repo command.

**Comment by isc-tleavitt** [comment]
Related to #481

**Comment by isc-kiyer** [comment]
> [@isc-kiyer](https://github.com/isc-kiyer) if you recall: have you primarily encountered this issue with the Perforce-based repository? (I saw some issues like this in my testing last week.)

@isc-tleavitt yes that is primarily where I have seen this but also in file system repos. So if module.xml ever changes, we need to update the cache for the repos before installing to ensure the cache isn't out of date.
